user/useradm.go

```diff
@@ -150,14 +162,15 @@ func (u *UserAdm) Login(ctx context.Context, email, pass string) (*jwt.Token, er
 	}
 
 	//get user
-	user, err := u.db.GetUserByEmail(ctx, email)
-
+	now := time.Now()
+	user, err := u.db.GetUserForLogin(ctx, email, now)
 	if user == nil && err == nil {
 		return nil, ErrUnauthorized
```

Comment:
just wondering aloud (feel free to ignore) we actually check if the login was successful, if it was not (the password mismatched as in the brute force attempt) we return the Unauthorized immediately. This will hint the attacker if the password was correct. dont you think that this does not exactly answer the "rate limits" requirement? (also I believe we should base rate limits on redis).

Reply:
also: with this patch I can lock-out any user knowing the email, by just trying to login. we need per ip limits and per user.

Reply:
In that case we should rate limit at the gateway level and offload it from the backend. Do you think I should close this PR and create a new one to mender-helm?